vendor/views/vendor_admin.py

```diff
@@ -183,4 +197,101 @@ def post(self, request):
+    '''
+    template_name = "vendor/manage/receipt_list.html"
+    model = Receipt
+
+    def get_queryset(self):
+        return self.model.objects.filter(products__in=Product.on_site.all(), order_item__offer__terms__lt=TermType.PERPETUAL)
+
+
+class AdminSubscriptionDetailView(LoginRequiredMixin, DetailView):
+    '''
+    Gets all Customer Profile information for quick lookup and management
+    '''
+    template_name = 'vendor/manage/subscription_detail.html'
+    model = Receipt
+    slug_field = 'uuid'
+    slug_url_kwarg = 'uuid'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        payment = Payment.objects.get(transaction=context['object'].transaction)
+
+        context['payment'] = payment
+        context['payment_form'] = CreditCardForm(initial={'payment_type': PaymentTypes.CREDIT_CARD})
+        context['billing_form'] = AddressForm(instance=payment.billing_address)
+
+        return context
+
+
+class AdminProfileListView(LoginRequiredMixin, ListView):
+    """
+    List of CustomerProfiles on site
+    """
+    template_name = "vendor/manage/profile_list.html"
+    model = CustomerProfile
+    queryset = CustomerProfile.on_site.all()
+
+
+class AdminProfileDetailView(LoginRequiredMixin, DetailView):
+    '''
+    Gets all Customer Profile information for quick lookup and management
+    '''
+    template_name = 'vendor/manage/profile_detail.html'
+    model = CustomerProfile
+    slug_field = 'uuid'
+    slug_url_kwarg = 'uuid'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['receipts'] = self.object.receipts.filter(products__in=Product.on_site.all(), order_item__offer__terms__gte=TermType.PERPETUAL)
+        context['subscriptions'] = self.object.receipts.filter(products__in=Product.on_site.all(), order_item__offer__terms__lt=TermType.PERPETUAL)
+
+        # context['products'] = [ product for receipt in self.object.receipts.all() for product in receipt.products.all() ]
+
+        return context
+
+
+class VoidProductView(LoginRequiredMixin, View):
+    success_url = reverse_lazy('vendor_admin:manage-profiles')
+
+    def post(self, request, *args, **kwargs):
+        receipt = Receipt.objects.get(uuid=self.kwargs["uuid"])
+        receipt.end_date = timezone.now()
+        receipt.save()
+
+        messages.info(request, _("Customer has no longer access to Product"))
+        return redirect(request.META.get('HTTP_REFERER', self.success_url))
+
+
+class AddOfferToProfileView(LoginRequiredMixin, View):
+
+    def get(self, request, *args, **kwargs):
+        customer_profile = CustomerProfile.objects.get(uuid=kwargs.get('uuid_profile'))
+        offer = Offer.objects.get(uuid=kwargs['uuid_offer'])
+
+        cart = customer_profile.get_cart_or_checkout_cart()
+        if cart.order_items.all().count():
+            messages.info(request, _("You must empty Cart"))
```

Comment:
Not sure I understand emptying the cart in the context of an admin.

Reply:
Actually I am going to change this to clear the cart. As you mention an admin those not know what's in the cart and should not care. The admin wants to give a offer to the user.